java-frontend/src/main/java/org/sonar/plugins/java/api/semantic/SymbolMetadata.java

```diff
@@ -60,8 +66,22 @@ public interface SymbolMetadata {
    */
   NullabilityData nullabilityData();
 
+  /**
+   * @deprecated use {@link #nullabilityData())} instead,
+   * this method uses the annotation of type parameter to compute nullability which is not correct
+   */
+  @Deprecated(since = "8.19.0")
```

Comment:
Let's add some more context and maybe explicitly set `forRemoval = true` to warn users to really not use this.